### DISCUSSION TITLE
Requirement of 'username'

### DISCUSSION THREAD

**Discussion by rowanarts** [xlab-si/xopera-opera#217] [Q&A] [answered]
Hi,

I want to deploy an image locally (from dockerhub) and I get the following error:
```
[Worker_0]   Deploying my-workstation_0
[Worker_0]   Deployment of my-workstation_0 complete
[Worker_0]   Deploying docker-private-registry_0
[Worker_0]     Executing create on docker-private-registry_0
Cannot find property 'username'.
```
I use the modules from [SODALITE/iac-modules](https://github.com/SODALITE-EU/iac-modules). 
And I'm trying to run the attached service.yml file by:
```
opera deploy service.yml --inputs input.yml -c
```

I use these files:
[input_and_service.zip](https://github.com/xlab-si/xopera-opera/files/7127430/input_and_service.zip)

note: input.yml.tmpl --> input.yml by adding docker_user & docker_password

**Question:**
Where is username required (cant find it in modules>docker or the playbooks that are used)?

**Comment by rowanarts** [comment]
I've exported everything into a CSAR, except for:
- I've removed the ca.crt & ca.key files that are required in the docker>artifacts folder.
- the input.yml file requires docker-user: <user> and docker-password: <password>.

[opera-package-5027f0.zip](https://github.com/xlab-si/xopera-opera/files/7128633/opera-package-5027f0.zip)

**Comment by anzoman** [comment, marked as answer]
Hi, @N8TFly and thanks for starting this discussion. To me it seems that the problem orignates from here: https://github.com/SODALITE-EU/iac-modules/blob/master/docker/docker_registry.yaml#L28, where `opera` tries to retrieve the `username` property from (I assume) node's `host` requirement, which is not directly defined within the `sodalite.nodes.DockerRegistry` node.

**Reply by rowanarts** [reply]
Looks like removing the `ansible_user` resolved the problem, thanks!
